src/components/Balance.tsx

```diff
@@ -32,10 +40,10 @@ export const Balance = ({
     };
 
     Object.values(allBalances).map((balance) => {
-      if (balance.token.type === "native") {
-        result.native = [...result.native, balance as Types.NativeBalance];
+      if (balance.assetType === "native") {
```

Comment:
would it be helpful to have 
```
enum ASSET_TYPE {
  native = "native"
}
```
and check the equality against ASSET_TYPE.native instead? I see you're checking if something is `=== "native"` a few times.